### PR TITLE
Feat/app-speech-synthesis

### PR DESCRIPTION
**Pull request by nestarz**
Add speech synthesis as an app.

## Breaking change

Any slackCommand should follow this format:
`<slackcommand> <parameter>`

Where `<slackcommand>` is **space-free**. 

For example, with a slackCommand defined as:

The following patterns are considered violations:

```
start daily
```

The following patterns are not considered violations:

```
start-daily
```

## Related Issue
https://github.com/CodeCorico/palantir/issues/20

### CODE REVIEW

File: src/tasks/views/Task.vue [hunk 1]
```diff
@@ -36,16 +36,16 @@ export default {
...
     if(this.slackCommand) {
-      const socket = io.connect('/')
-      socket.on(this.slackCommand, this.start)
+      const socket = io.connect('/');
+      socket.on(this.slackCommand, parameters => this.start(parameters));
     }
   },
   methods: {
-    start() {
+    start(slackParameters = undefined) {
```
Comment (XavierBoubert) [inline src/tasks/views/Task.vue]: The default is already `undefined`

File: src/server/index.js [hunk 2]
```diff
@@ -55,21 +56,55 @@ app.use((req, res) => {
...
+      rtm.on('message', async ({ text, user, channel }) => {
+        if (!text) {
+          return;
         }
+
+        const [command, parameters] = text.split(/ (.*)/).filter(e => e.length > 1);
+
+        if (apps.indexOf(command) === -1) {
+          if (command === 'help') {
+            rtm.sendMessage(`<@${user}>, voici les commandes disponibles: ${
```
Comment (XavierBoubert) [inline src/server/index.js]: The texts have to be in english

File: src/app-speech-synthesis/stores/speech-synthesis.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+      const { data } = await axios.get(`${SPEECH_SYNTHESIS_API_URL}?${query}`);
+
+      if (data.error) {
+        // eslint-disable-next-line no-console
+        console.error(data.error);
+
+        if (slack) {
+          socket.emit('slackMessage', [
+            `*App(${name})* `,
+            `<@${slack.user}>, Désolé, une erreur s'est produite.\n`,
```
Comment (XavierBoubert) [inline src/app-speech-synthesis/stores/speech-synthesis.js]: The texts have to be in english

File: src/tasks/views/Task.vue [hunk 4]
```diff
@@ -58,6 +58,7 @@ export default {
       this.$store.dispatch(this.dispatch, {
         id: this.id,
         config: this.config,
+        __slack: slackParameters
```
[the remark this answers is not in the document]
Comment (nestarz) [reply src/tasks/views/Task.vue]: Done !

File: src/tasks/views/Task.vue (hunk 1 above)
Comment (nestarz) [reply src/tasks/views/Task.vue]: Done !

File: src/server/index.js [hunk 5]
```diff
@@ -88,5 +128,4 @@ if (statics) {
 
 // eslint-disable-next-line no-console
 const server = app.listen(port, () => console.log(`Server started on :${port}`));
-const io = socket(server)
-
+const io = socket(server);
```
[the remark this answers is not in the document]
Comment (nestarz) [reply src/server/index.js]: Done !

File: src/server/index.js (hunk 2 above)
Comment (nestarz) [reply src/server/index.js]: Done !

File: src/server/index.js [hunk 6]
```diff
@@ -55,21 +56,55 @@ app.use((req, res) => {
...
+        }
+
+        const response = await web.users.info({ user });
+        io.emit(command, {
+          parameters,
+          user: response.user.profile.display_name,
+          channel,
+        });
+        rtm.sendMessage([
+          'C\'est parti !!\n',
```
[the remark this answers is not in the document]
Comment (nestarz) [reply src/server/index.js]: Done !

File: src/app-speech-synthesis/stores/speech-synthesis.js (hunk 3 above)
Comment (nestarz) [reply src/app-speech-synthesis/stores/speech-synthesis.js]: Done !

**Pull request merged**